app/models/Contribution.java

```diff
@@ -1338,6 +1341,25 @@ public void setCampaignIds(List<Long> cids) {
         }
     }
 
+    public List<Long> getcontainingContributionsIds() {
+        containingContributionsIds = new ArrayList<>();
+        for (ResourceSpace resourceSpace : this.containingSpaces) {
+            if(resourceSpace.getType() == ResourceSpaceTypes.CONTRIBUTION){
+                Contribution c = this.findByResourceSpaceId(resourceSpace.getResourceSpaceId());
+                containingContributionsIds.add(c.getContributionId());
+            }
+        }
+        return containingContributionsIds;
+    }
+
+    public void setContainingContributionsIds(List<Long> cids) {
```

Comment:
@MarceMMAD let's remove the setter as this might end overriding the relationship between contributions and its containers from the frontend. We don't want that.